gcampus/core/tasks.py

```diff
@@ -0,0 +1,214 @@
+    with transaction.atomic():
+        for access_key in old_access_keys:
+            access_key.deactivated = True
+            access_key.save()
+            access_key_count += 1
+            courses.setdefault(access_key.course_id, [])
+            courses[access_key.course_id].append(access_key.token)
+
+    # Send email notifying the teacher about deactivated courses
+    for course_id, access_keys in courses.items():
+        course = Course.objects.only("teacher_email", "name", "school_name").get(
+            pk=course_id
+        )
+        send_access_key_deactivation_email.apply_async(
+            args=(course.teacher_email, course.name, course.school_name, access_keys)
+        )
+
+    mail_managers(
+        "Maintenance report",
+        f"Environment: {getattr(settings, 'ENVIRONMENT', 'not set')}\n\n"
+        "Changes:\n"
+        f"Measurements marked as deleted: {measurement_count}\n"
+        f"Unverified courses deleted: {unverified_courses_count}\n"
+        f"Unused courses deleted: {unused_courses_count}\n"
+        "Old access keys deactivated: "
+        f"{access_key_count} out of {len(courses)} courses\n\n"
+        f"(Version: {settings.VERSION}, {timezone.now().isoformat()}",
+        fail_silently=True,
+    )
+
+
+@shared_task
+def staging_maintenance():
+    now = timezone.now()
+    if getattr(settings, "ENVIRONMENT", None) != "dev":
+        logger.error(
+            "Task 'staging_maintenance' has been run outside the staging environment"
+        )
+        return
+
+    # Delete all hidden measurements
+    m_hidden = Measurement.all_objects.filter(hidden=True)
+    m_count = m_hidden.all.count()
+    m_hidden.delete()
+
+    # Delete all old measurements
+    m_old = Measurement.objects.filter(
+        updated_at__lt=now - settings.MEASUREMENT_LIFETIME_STAGING
+    )
+    m_count += m_old.all().count()
+    m_old.delete()
+
+    # Delete all old access keys
+    course_deletion_date = now - settings.COURSE_LIFETIME_STAGING
+    a = AccessKey.objects.filter(
+        Q(last_login__isnull=True) | Q(last_login__lt=course_deletion_date)
+    )
+    a_count = a.all().count()
+    a.delete()
+
+    # Delete all old courses
+    c = Course.objects.filter(
+        Q(access_keys__isnull=True),
+        # AND
+        Q(course_token__last_login__isnull=True)
+        | Q(course_token__last_login__lt=course_deletion_date),
+    )
+    c_count = c.all().count()
+    c.delete()
+
+    mail_managers(
+        "'dev' environment maintenance report",
+        f"Environment: {getattr(settings, 'ENVIRONMENT', 'not set')}\n\n"
+        "Changes:\n"
+        f"Measurements deleted: {m_count:d}\n"
+        f"Old access keys deleted: {a_count}\n"
+        f"Old courses deleted: {c_count}\n\n"
+        f"(Version: {settings.VERSION}, {timezone.now().isoformat()}",
+        fail_silently=True,
+    )
+
+
+@shared_task
+def refresh_water_from_osm():
+    now = timezone.now()
+    waters: List[Water] = Water.objects.filter(
```

Comment:
Do we need to order the waters by the date on which they have been updated before selecting the first ones?